src/controllers/midi/midicontroller.cpp

```diff
@@ -537,3 +585,36 @@ void MidiController::processInputMapping(const MidiInputMapping& mapping,
     qCWarning(m_logBase) << "MidiController: No script function specified for"
                          << MidiUtils::formatSysexMessage(getName(), data, timestamp);
 }
+
+const QString sErrorMessage(
+        "Invalid timer callback provided to midi.makeInputHandler. "
+        "Please pass a function and make sure that your code contains no syntax errors.");
+
+void MidiController::makeInputHandler(int status, int midino, const QJSValue& scriptCode) {
+    if (!scriptCode.isCallable()) {
+        auto error = sErrorMessage;
+        if (scriptCode.isError()) {
+            error.append("\n" + scriptCode.toString());
+        }
+        this->getScriptEngine()->throwJSError(error);
+        return;
+    }
+
+    if (status <= 0 || midino <= 0) {
+        auto mStatusError = std::format(
+                "Invalid status or midino passed to midi.makeInputHandler. "
+                "Please pass a strictly positive integer. status={},midino={}",
+                status,
+                midino);
```

Comment:
as much as I like std::format too, we still support compilers that don't have a built-in implementation. So you'll have to resort to Qt's formatting facilities for now.

```suggestion
        auto mStatusError = QStringLiteral(
                "Invalid status or midino passed to midi.makeInputHandler. "
                "Please pass a strictly positive integer. status=%1,midino=%2").arg(
                status,
                midino);
```